src/Select/Select.jsx

```diff
@@ -44,6 +44,22 @@ export function Select({
 }) {
   const {cssClass} = Select;
 
+  if (!lazy) {
```

Comment:
An alternative, in that case, is to do it at the type checking phase with custom validators:

```
import { all } from "prop-types-extra";
// ...
// validator to ensure props are not truthy at the same time
function numTruthy(num, ...propNames) {
  return (props) => 
    propNames.filter(name => !!props[name]) !== num ?
      new Error(`Exactly ${num} of props ${JSON.stringify(propNames)} must be specified") : null
}

// custom validator to ensure props are all truthy, or are all falsey
function allOrNoneTruthy(...propNames) {
  return (props) => {
      [0, propNames.length].indexOf(propNames.filter(name => !!props[name])) !== -1 ?
        null : new Error(`Props ${JSON.stringify(propNames)} must be specified together") :
}

Select.propTypes = {
  // ...
  options: all(
    React.PropTypes.arrayOf(selectValuePropType),
    numTruthy(1, "lazy", "options"),
  ),
  lazy: all(
    React.PropTypes.bool,
    numTruthy(1, "lazy", "options"),
    allOrNoneTruthy("lazy", "loadOptions"),
  ),
  loadOptions: all(
    React.PropTypes.func,
    allOrNoneTruthy("lazy", "loadOptions"),
  )
  // ...
}
```

But if this is something you don't like, then the current approach is OK.